tests/Runner/Parallel/ProcessFactoryTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Runner\Parallel;
+
+use PhpCsFixer\Console\Command\FixCommand;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Runner\Parallel\ParallelConfig;
+use PhpCsFixer\Runner\Parallel\ProcessFactory;
+use PhpCsFixer\Runner\Parallel\ProcessIdentifier;
+use PhpCsFixer\Runner\RunnerConfig;
+use PhpCsFixer\Tests\TestCase;
+use PhpCsFixer\ToolInfo;
+use React\EventLoop\StreamSelectLoop;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputDefinition;
+
+/**
+ * @author Greg Korba <greg@codito.dev>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Runner\Parallel\ProcessFactory
+ */
+final class ProcessFactoryTest extends TestCase
+{
+    private InputDefinition $inputDefinition;
+
+    protected function setUp(): void
+    {
+        $fixCommand = new FixCommand(new ToolInfo());
+        $application = new Application();
+        $application->add($fixCommand);
+
+        // In order to have full list of options supported by the command (e.g. `--verbose`)
+        $fixCommand->mergeApplicationDefinition(false);
+
+        $this->inputDefinition = $fixCommand->getDefinition();
+    }
+
+    /**
+     * @requires OS Linux|Darwin
```

Comment:
potentially blocking

how will parallel execution behave on Windows?
if it would not work, let's have a check when enabling it under windows to crash and complain

Reply:
The cause of this [lays in `react/child-process` component](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/actions/runs/8105416319/job/22153708606?pr=7777#step:7:2090)

> 1) PhpCsFixer\Tests\Console\Command\WorkerCommandTest::testWorkerCommunicatesWithTheServer
> Process pipes are not supported on Windows due to their blocking nature on Windows

But at the same time [running Fixer in parallel works on Windows](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/actions/runs/8903582120/job/24451564940?pr=7777#step:14:19), so it's rather not about parallel runner itself, but about how tests are created. Maybe I did something wrong and just couldn't work around it. @WyriHaximus do you know why I got that error, since React processes are created with explicit `$fds`:

```php
$this->process = new ReactProcess($this->command, null, null, [
    1 => $this->stdOut,
    2 => $this->stdErr,
]);
```

(`stdOut` and `stdErr` are created with `tmpfile()`).

Reply:
if you can put it as a comment to test, would be awesome.

then, non-actionable for me (if we enable this test one day, awesome. if not, OKish)